machine.go

```diff
@@ -1067,3 +1071,111 @@ func (m *Machine) CreateSnapshot(ctx context.Context, memFilePath, snapshotPath
 	m.logger.Debug("snapshot created successfully")
 	return nil
 }
+
+// setupBalloonDevice set up a balloon device if one does not exist
+func (m *Machine) CreateBalloonDevice(ctx context.Context, dev models.Balloon) error {
+	fmt.Println("Start to create a balloon device")
+	balloonConfig := models.Balloon{
+		AmountMb: Int64(0),
+		DeflateOnOom: Bool(false),
+		StatsPollingIntervals: 0,
+	}
+
+	if dev.AmountMb != nil {
+		balloonConfig.AmountMb = dev.AmountMb
+	}
+
+	if dev.DeflateOnOom != nil {
+		balloonConfig.DeflateOnOom = dev.DeflateOnOom
+	}
+
+	if dev.StatsPollingIntervals != 0 {
+		balloonConfig.StatsPollingIntervals = dev.StatsPollingIntervals
+	}
+
+	amountMb := *(balloonConfig.AmountMb)
+	m.logger.Infof("Install a balloon device with %v MB", amountMb)	
+	respNoContent, err := m.client.PutBalloon(ctx, &balloonConfig)
+
+	if err == nil {
+		m.logger.Printf("Installed balloon device with %v MB: %s", amountMb, respNoContent.Error())
+	} else {
+		m.logger.Errorf("Install balloon device failed : %s", err)
+	}
+	return err
+}
+
+// GetBalloonConfig gets the current balloon device configuration.
+func (m *Machine) GetBalloonConfig(ctx context.Context, v interface{}) error {
+	resp, err := m.client.DescribeBalloonConfig(ctx)
+	if err != nil {
+		m.logger.Errorf("Getting balloonConfig: %s", err)
+		return err
+	}
+
+	payloadData, err := json.Marshal(resp.Payload)
+	if err != nil {
+		m.logger.Errorf("Getting balloonConfig failed parsing payload: %s", err)
+		return err
+	}
+
+	if err := json.Unmarshal(payloadData, v); err != nil {
+		m.logger.Errorf("Getting balloonConfig failed parsing payload: %s", err)
+		return err
+	}
+
+	m.logger.Printf("GetBalloonConfig successful")
+	return nil
+}
+
+// UpdateBalloon will modify the current balloon with the new 
+// parameters of the ballonUpdate
```

Comment:
We can also add an example
```go
// UpdateBalloon will modify the current balloon with the new 
// parameters of the ballonUpdate
// Example:
//  machine := firecrack.NewMachine(
//		context.Background(),
//		Config{
//			DisableValidation: true,
//			MachineCfg: models.MachineConfiguration{
//				VcpuCount:   Int64(1),
//				MemSizeMib:  Int64(100),
//				CPUTemplate: models.CPUTemplate(models.CPUTemplateT2),
//				HtEnabled:   Bool(false),
//			},
//		})
//	machine.FcInit = machine.FcInit.Append(firecracker.CreateBalloonConfigHandler)
//	machine.UpdateBalloon(ctx, 256)
```

Reply:
Also add a `CreateBalloonOpt` that automatically injects the handler

Reply:
I added `PatchBalloonOpt`